core/src/main/java/com/linecorp/armeria/common/logging/ContentPreviewerFactory.java

```diff
@@ -0,0 +1,191 @@
+        final Map<MediaType, Supplier<ContentPreviewer>> maps = new HashMap<>();
+        for (MediaType type : contentTypes) {
+            maps.put(type, supplier);
+        }
+        return of(maps);
+    }
+
+    /**
+     * Creates a new instance of {@link ContentPreviewerFactory} which creates a {@link ContentPreviewer}
+     * through {@code supplier} if a request/response mathces any of {@code contentTypes}.
+     */
+    static ContentPreviewerFactory of(Supplier<ContentPreviewer> supplier, MediaType... contentTypes) {
+        return of(supplier, Arrays.asList(contentTypes));
+    }
+
+    /**
+     * Creates a new instance of {@link ContentPreviewerFactory} which creates a {@link ContentPreviewer}
+     * through {@code supplier} if the content type of a request/response mathces any of {@code contentTypes}.
+     */
+    static ContentPreviewerFactory of(Supplier<ContentPreviewer> supplier, String... contentTypes) {
+        return of(supplier, Arrays.stream(contentTypes).map(MediaType::parse).collect(Collectors.toList()));
+    }
+
+    /**
+     * Creates a new instance of {@link ContentPreviewerFactory} creating a {@link ContentPreviewer}
+     * which produces the text with the maximum {@code length}
+     * if the content type of a request/response mathces any of {@code contentTypes}.
+     */
+    static ContentPreviewerFactory ofText(int length, Charset defaultCharset,
+                                          Iterable<MediaType> contentTypes) {
+        return of(() -> ContentPreviewer.ofText(length, defaultCharset), contentTypes);
+    }
+
+    /**
+     * Creates a new instance of {@link ContentPreviewerFactory} creating a {@link ContentPreviewer}
+     * which produces the text with the maximum {@code length}
+     * if the content type of a request/response mathces any of {@code contentTypes}.
+     */
+    static ContentPreviewerFactory ofText(int length, Charset defaultCharset, MediaType... contentTypes) {
+        return ofText(length, defaultCharset, Arrays.asList(contentTypes));
+    }
+
+    /**
+     * Creates a new instance of {@link ContentPreviewerFactory} creating a {@link ContentPreviewer}
+     * which produces the text with the maximum {@code length} limit
+     * if the content type of a request/response matches any of {@code contentTypes}.
+     */
+    static ContentPreviewerFactory ofText(int length, Charset defaultCharset, String... contentTypes) {
+        return ofText(length, defaultCharset, Arrays.stream(contentTypes).map(MediaType::parse).collect(
+                Collectors.toList()));
+    }
+
+    /**
+     * Creates a new instance of {@link ContentPreviewerFactory} creating a {@link ContentPreviewer}
+     * which produces the text with the maximum {@code length} limit
+     * if the content type of a request/response matches
+     * any of {@link ArmeriaHttpUtil#HTTP_TEXTIBLE_MEDIA_TYPES}.
```

Comment:
`ArmeriaHttpUtil` is not part of the public API. You need to list them all explicitly for users.